include/openenclave/internal/defs.h

```diff
@@ -7,8 +7,21 @@
 #include <openenclave/bits/defs.h>
 
 /* OE_WEAK_ALIAS */
+#ifdef __GNUC__
 #define OE_WEAK_ALIAS(OLD, NEW) \
     extern __typeof(OLD) NEW __attribute__((__weak__, alias(#OLD)))
+#elif _MSC_VER
+#define OE_WEAK_ALIAS __pragma(comment(linker, "/alternatename:" #NEW "=" #OLD))
+#else
+#error OE_WEAK_ALIAS not implemented
+#endif
+
+/* OE_WEAK */
+#ifdef __GNUC__
+#define OE_WEAK __attribute__((weak))
+#else
+#define OE_WEAK
```

Comment:
This should probably error out as in the `OE_WEAK_ALIAS` case if used in a compiler context that does not support it.

Reply:
Honestly I just couldn't figure out how to do it. `#error` will always error when compiler is msvc. Is there some mechanism we use elsewhere that will throw an error when a macro is used?

Reply:
https://stackoverflow.com/questions/24225006/warning-and-error-as-macro

Reply:
_STATIC_ASSERT(FALSE)
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/static-assert-macro?view=vs-2019
https://docs.microsoft.com/en-us/cpp/cpp/static-assert?view=vs-2019